frontend/src/features/mapSideBar/MotiInventoryContainer.tsx

```diff
@@ -34,163 +29,180 @@ export interface IMotiInventoryContainerProps {
 
   const {
     getPropertyAssociations,
     isLoading: propertyAssociationsLoading,
   } = usePropertyAssociations();
 
+  const { getLtsaData, ltsaLoading } = useLtsa();
+
+  const [composedProperty, setComposedProperty] = useState<ComposedProperty>({
+    apiPropertyLoading: ltsaLoading,
+    ltsaLoading: propertyLoading,
+    propertyAssociationsLoading: propertyAssociationsLoading,
+  });
+
+  useEffect(() => {
+    setComposedProperty(property => ({
+      ...property,
+      pid: props.pid,
+    }));
+    setIsEditing(false);
+  }, [props.pid]);
+
+  useEffect(() => {
+    setComposedProperty(property => ({
+      ...property,
+      ltsaLoading: ltsaLoading,
+      apiPropertyLoading: propertyLoading,
+      propertyAssociationsLoading: propertyAssociationsLoading,
+    }));
+  }, [propertyLoading, ltsaLoading, propertyAssociationsLoading]);
+
+  const fetchPimsProperty = React.useCallback(async () => {
+    try {
+      if (!!props.pid) {
+        const propInfo = await getPropertyWithPid(props.pid);
+        if (isMounted() && propInfo.pid === pidFormatter(props.pid)) {
+          setComposedProperty(property => ({ ...property, apiProperty: propInfo }));
+        }
+      }
+    } catch (e) {
+      // PSP-2919 Hide the property info tab for non-inventory properties
+      // We get an error because PID is not on our database
+      if (axios.isAxiosError(e)) {
+        const axiosError = e as AxiosError<IApiError>;
+        if (axiosError?.response?.status === 404) {
+          setComposedProperty(property => ({ ...property, apiProperty: undefined }));
+        }
+      }
+    }
+  }, [getPropertyWithPid, isMounted, props.pid]);
+
   useEffect(() => {
-    async function fetchResearchFile() {
+    fetchPimsProperty();
+  }, [fetchPimsProperty]);
+
+  useEffect(() => {
+    const getAssociations = async () => {
       if (props.pid !== undefined) {
         const response = await getPropertyAssociations(props.pid);
-        setPropertyAssociations(response);
+        if (response !== undefined) {
+          setComposedProperty(property => ({ ...property, propertyAssociations: response }));
+        }
       }
-    }
-    fetchResearchFile();
+    };
+    getAssociations();
   }, [getPropertyAssociations, props.pid]);
 
-  // After API property object has been received, we query relevant map layers to find
-  // additional information which we store in a different model (IPropertyDetailsForm)
-  const propertyViewForm = usePropertyDetails(apiProperty);
-
-  const { getLtsaData, ltsaLoading } = useLtsa();
   useEffect(() => {
     const func = async () => {
-      setLtsaDataRequestedOn(new Date());
-      setLtsaData(undefined);
+      setComposedProperty(property => ({
+        ...property,
+        ltsaDataRequestedOn: new Date(),
+        ltsaData: undefined,
+      }));
+
       if (!!props.pid) {
         const ltsaData = await getLtsaData(pidFormatter(props.pid));
```

Comment:
my pr updates a lot of these to use the id instead of the pid, but I don't mind if you go first and I resolve afterwords.

Reply:
I rather that. This one gave me a lot of headache when updating the tests